File: apiGateway/controllers/businessControllers/businessController.js
const axios = require("axios");
const https = require("https");

const agent = new https.Agent({ rejectUnauthorized: false });

async function getAllBusinesses(req, res) {
  try {
    const result = await axios.get("https://localhost:7218/business/allBusinesses", { httpsAgent: agent });
    res.json(result.data);
  } catch (error) {
    console.error(error);
    res.status(500).json({ error: "Internal Server Error" });
  }
}

async function getBusiness(req, res) {
  try {
    const id = req.params.id;
    const result = await axios.get(`https://localhost:7218/business/${id}`, { httpsAgent: agent });
    res.json(result.data);
  } catch (error) {
    console.error(error);
    res.status(500).json({ error: "Internal Server Error" });
  }
}

async function createBusiness(req, res) {
  try {
    const existingBusiness = await axios.get(`https://localhost:7218/business/${req.body.businessID}`, { httpsAgent: agent });
    if (existingBusiness.data) {
      return res.status(409).json({ error: "Business already exists." });
    }

    const result = await axios.post("https://localhost:7218/business/createBusiness", req.body, { httpsAgent: agent });
    res.status(201).json({ message: "Successfully created." });
  } catch (error) {
    console.error(error);
    res.status(500).json({ error: "Internal Server Error" });
  }
}

async function updateBusiness(req, res) {
  try {
    const id = req.params.id;
    if (id !== req.body.businessID) {
      return res.status(400).json({ error: "Business ID mismatch." });
    }

    const existingBusiness = await axios.get(`https://localhost:7218/business/${id}`, { httpsAgent: agent });
    if (!existingBusiness.data) {
      return res.status(404).json({ error: "Business not found." });
    }

    await axios.put(`https://localhost:7218/business/updateBusiness/${id}`, req.body, { httpsAgent: agent });
    res.json({ message: "Successfully updated." });
  } catch (error) {
    console.error(error);
    res.status(500).json({ error: "Internal Server Error" });
  }
}

async function deleteBusiness(req, res) {
  try {
    const id = req.params.id;
    const existingBusiness = await axios.get(`https://localhost:7218/business/${id}`, { httpsAgent: agent });
    if (!existingBusiness.data) {
      return res.status(404).json({ error: "Business not found." });
    }

    await axios.delete(`https://localhost:7218/business/deleteBusiness/${id}`, { httpsAgent: agent });
    res.json({ message: "Successfully deleted." });
  } catch (error) {
    console.error(error);
    res.status(500).json({ error: "Internal Server Error" });
  }
}

module.exports = {
  getAllBusinesses,
  getBusiness,
  createBusiness,
  updateBusiness,
  deleteBusiness,
};

File: businessService/businessService/Controllers/businessController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using businessService.Models;
using businessService.Repositories.Business;

using Microsoft.AspNetCore.Mvc;

namespace businessService.Controllers
{
    [Route("[controller]")]
    [ApiController]
    public class businessController : ControllerBase
    {
        private readonly IBusinessRepository _businessRepository;

        public businessController(IBusinessRepository businessRepository)
        {
            _businessRepository = businessRepository;
        }

        [HttpGet("allBusinesses")]
        public async Task<IEnumerable<BusinessModel>> GetAllBusinesses()
        {
            return await _businessRepository.getAllBusinesses();
        }

        [HttpGet("{id}")]
        public async Task<ActionResult<BusinessModel>> GetBusiness(string id)
        {
            var business = await _businessRepository.getBusiness(id);
            if (business == null)
            {
                return NotFound();
            }
            return business;
        }

        [HttpPost("createBusiness")]
        public async Task<IActionResult> CreateBusiness([FromBody] BusinessModel business)
        {
            var businessToAdd= await _businessRepository.getBusiness(business.businessID);
            if (businessToAdd != null)
            {
                return Conflict("Business already exists.");
            }
            var newBusiness = await _businessRepository.createBusiness(business);
            if (newBusiness == null)
            {
                return StatusCode(500, "An error occurred while creating the business.");
            }
            else
            {
                return Ok("Successfully created.");
            }
        }


        [HttpPut("updateBusiness/{id}")]
        public async Task<IActionResult> UpdateBusiness(string id, [FromBody] BusinessModel business)
        {
            if (id != business.businessID)
            {
                return BadRequest();
            }
            var existingBusiness = await _businessRepository.getBusiness(id);
            if (existingBusiness == null)
            {
                return NotFound("Business not found.");
            }
            try
            {
                await _businessRepository.updateBusiness(business);
                return Ok("Successfully updated.");
            }
            
            catch (Exception)
            {
                return StatusCode(500, "An error occurred while updating the business.");
            }
        }


        [HttpDelete("deleteBusiness/{id}")]
        public async Task<IActionResult> DeleteBusiness(string id)
        {
            var businessToDelete = await _businessRepository.getBusiness(id);
            if (businessToDelete == null)
            {
                return NotFound("Business not found.");
            }

            await _businessRepository.deleteBusiness(id);
            return Ok("Successfully deleted.");
        }
    }
}
